src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String OPERATION_TYPE_BUY = "buy";
+    private static final String OPERATION_TYPE_SUPPLY = "supply";
+    private static final String COMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        StringBuilder report = new StringBuilder();
+        List<String> stringFromFile = new ArrayList<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String read = bufferedReader.readLine();
+            while (read != null) {
+                stringFromFile.add(read);
+                read = bufferedReader.readLine();
+            }
+            buildReport(stringFromFile, report);
+
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to read from file: " + fromFileName, e);
+        }
+        writeToFile(toFileName, report);
+    }
```

Comment:
The `getStatistic` method seems to be doing too much. It's reading from a file, processing data, and writing to another file. Consider dividing the program logic into separate methods to handle each task individually, which will make the code easier to read and maintain.